tests/unit/components/sl-date-range-picker-test.js

```diff
@@ -0,0 +1,352 @@
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-end-date input' ).prop( 'placeholder' ),
+        '',
+        'End date input placeholder is empty when endDatePlaceholder is undefined'
+    );
+});
+
+test( 'label is accepted as a parameter', function( assert ) {
+    let labelText = 'lorem ipsum';
+    let component = this.subject({ label: labelText });
+
+    assert.equal(
+        component.get( 'label' ),
+        labelText,
+        'label was accepted as a parameter'
+    );
+
+    assert.equal(
+        this.$( 'label' ).html(),
+        labelText,
+        'label element was created with label parameter text'
+    );
+
+    assert.equal(
+        this.$( 'label' ).prop( 'for' ),
+        component.get( 'inputElementId' ),
+        'label element has the correct for property'
+    );
+
+    assert.strictEqual(
+        component.get( 'startDatePlaceholder' ),
+        undefined,
+        'startDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-start-date input' ).prop( 'placeholder' ),
+        '',
+        'Start date input placeholder is empty when startDatePlaceholder is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'endDatePlaceholder' ),
+        undefined,
+        'endDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-end-date input' ).prop( 'placeholder' ),
+        '',
+        'End date input placeholder is empty when endDatePlaceholder is undefined'
+    );
+});
+
+test( 'startDatePlaceholder is accepted as a parameter', function( assert ) {
+    let placeholderText = 'lorem ipsum';
+    let component = this.subject({ startDatePlaceholder: placeholderText });
+
+    assert.equal(
+        component.get( 'startDatePlaceholder' ),
+        placeholderText,
+        'startDatePlaceholder was accepted as a parameter'
+    );
+
+    assert.equal(
+        this.$( '.sl-daterange-start-date input' ).prop( 'placeholder' ),
+        placeholderText,
+        'Start date input placeholder was passed through correctly'
+    );
+
+    assert.strictEqual(
+        component.get( 'label' ),
+        undefined,
+        'label is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( 'label' )[0],
+        undefined,
+        'No label is created when label is undefined'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-end-date input' ).prop( 'placeholder' ),
+        '',
+        'End date input placeholder is empty when endDatePlaceholder is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'endDatePlaceholder' ),
+        undefined,
+        'endDatePlaceholder is undefined by default'
+    );
+});
+
+test( 'endDatePlaceholder is accepted as a parameter', function( assert ) {
+    let placeholderText = 'lorem ipsum';
+    let component = this.subject({ endDatePlaceholder: placeholderText });
+
+    assert.equal(

```

Comment:
This assertion is not needed, as it is testing Ember functionality and not that of this specific component.  A user of the component can pass in any properties by any name they desire.  Our code will only use the ones that it cares about.  What we specifically do with the specific properties we care about is what we write tests against.